src/safe-upgrade-tool/developer.md

```diff
@@ -0,0 +1,150 @@
+*  `vendor/bin/phpunit -c tests/unit/phpunit.xml.dist --testsuite=unit-tests`
+
+## Integration testing
+
+To run the integration tests, execute one of the following commands:
+
+*  `vendor/bin/phpunit -c tests/integration/phpunit.xml.dist tests/integration`
+*  `vendor/bin/phpunit -c tests/integration/phpunit.xml.dist --testsuite=integration-tests`
+
+## Acceptance testing
+
+1. Before executing acceptance tests, you must set the Magento URL in the `phpunit` configuration file.
+1. Copy the default `tests/acceptance/phpunit.xml` file (without the .dist suffix).
+1. Change the `TESTS_BASE_URL` value to point to the Magento URL that you want to test.
+1. To run the acceptance tests, execute one of the following commands:
+
+*  `vendor/bin/phpunit -c tests/acceptance/phpunit.xml tests/acceptance`
+*  `vendor/bin/phpunit -c tests/acceptance/phpunit.xml --testsuite=acceptance-tests`
+
+## JS unit testing for GraphQL
+
+We used the [Jest](https://jestjs.io/docs/en/getting-started.html) framework to create these JS unit tests:
+
+{:.bs-callout-info}
+To run JS unit tests for GraphQL, you must have Node.js installed.
+
+### Node.js
+
+To install Node.js on your system, see the Node.js [documentation](https://nodejs.dev/learn/how-to-install-nodejs).
+
+The following instructions are for MacOS systems:
+
+1. Open a terminal and navigate to the `graphql-schema-compatibility/` directory.
+1. Install project dependencies:
+
+   ```bash
+   npm install
+   ```
+
+1. Run unit tests or `jest`:
+
+   ```bash
+   npm run unit-test
+   ```
+
+The tests are inside `graphql-schema-compatibility/test/js/unit`.
+
+The string schemas for testing are inside `dev/graphql_schemas`.
+
+## Complexity score
+
+The **complexity score** is a figure that indicates how difficult an upgrade from the current version to the new one might be. Lower numbers indicate easier upgrades.
+
+{:.bs-callout-info}
+Complexity scores range between 0 and 100.
+
+This score is based on the results extracted from the analysis:
+
+*  Number of issues identified
+*  Severity of issues identified
+
+The SUT calculates this score according to the following formula:
+
+### Complexity score formula
+
+`Complexity Score = 2 * (# of errors) + 1 * (# of warnings)`
+
+## Packaging
+
+A package is essentially a directory containing something. It contains a package description which has a name and a version. The name and the version are used to identify the package.
+
+It is possible to create a `sut.zip` file of the SUT to distribute it with partners as a composer package.
+
+In order to upload it to be able to distribute it, you need first to create the package:
+
+1. To package the SUT execute the following command:
+
+   ```bash
+   bin/package VERSION`
+   ```
+
+### An example of a package
+
+`bin/package 0.0.1`
+
+1. This generates an `sut.zip` file that you can upload (or distribute with) to the [Magento Composer repository](https://repo.magento.com).
```

Comment:
```suggestion
1. This generates an `sut.zip` file that you can upload to the [Magento Composer repository](https://repo.magento.com).
```

Reply:
As far as I know, only authorized employees can upload packages to repo.magento.com. Are you sure this belongs in our public docs?

Reply:
will remove all packaging section then